dao/src/main/java/com/epam/esm/dao/impl/GiftCertificateDaoImpl.java

```diff
@@ -32,115 +26,117 @@
  */
 @Repository
 public class GiftCertificateDaoImpl implements GiftCertificateDao {
-    private final JdbcTemplate jdbcTemplate;
-    private final GiftCertificateMapper mapper;
+    private final SessionFactory sessionFactory;
 
     @Autowired
-    public GiftCertificateDaoImpl(JdbcTemplate jdbcTemplate, GiftCertificateMapper mapper) {
-        this.jdbcTemplate = jdbcTemplate;
-        this.mapper = mapper;
+    public GiftCertificateDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
     }
 
     @Override
-    public GiftCertificate add(GiftCertificate giftCertificate) {
-        KeyHolder keyHolder = new GeneratedKeyHolder();
-        String name = giftCertificate.getName();
-        String description = giftCertificate.getDescription();
-        BigDecimal price = giftCertificate.getPrice();
-        Integer duration = giftCertificate.getDuration();
-        LocalDateTime createDate = giftCertificate.getCreateDate();
-        LocalDateTime lastUpdateDate = giftCertificate.getLastUpdateDate();
-        jdbcTemplate.update(con -> {
-            PreparedStatement preparedStatement = con.prepareStatement(GiftCertificateSql.ADD_GIFT_CERTIFICATE, Statement.RETURN_GENERATED_KEYS);
-            preparedStatement.setString(1, name);
-            preparedStatement.setString(2, description);
-            preparedStatement.setBigDecimal(3, price);
-            preparedStatement.setInt(4, duration);
-            preparedStatement.setTimestamp(5, Timestamp.valueOf(createDate));
-            preparedStatement.setTimestamp(6, Timestamp.valueOf(lastUpdateDate));
-            return preparedStatement;
-        }, keyHolder);
-        if (keyHolder.getKey() == null) {
-            throw new ResourceNotAddedException("GiftCertificate not Add.No KeyHolder", ExceptionCauseCode.GIFT_CERTIFICATE);
-        }
-        long giftCertificateId = keyHolder.getKey().longValue();
-        return new GiftCertificate(giftCertificateId, name, description, price, duration, createDate, lastUpdateDate);
+    public void add(GiftCertificate giftCertificate) {
+        Session session = sessionFactory.getCurrentSession();
+        session.persist(giftCertificate);
+        session.flush();
```

Comment:
Why do you use `session.flush();` here and nowhere else in the persist methods? Please, use it only if it's necessary and where it's necessary

Reply:
Removed flush